lib/SPIRV/SPIRVReader.cpp

```diff
@@ -2492,6 +2492,14 @@ Value *SPIRVToLLVM::transValueWithoutDecoration(SPIRVValue *BV, Function *F,
   case OpSignBitSet:
     return mapValue(BV,
                     transRelational(static_cast<SPIRVInstruction *>(BV), BB));
+  case OpIAddCarry: {
+    IRBuilder Builder(BB);
+    auto *BC = static_cast<SPIRVBinary *>(BV);
+    return mapValue(BV, Builder.CreateBinaryIntrinsic(
+                            Intrinsic::uadd_with_overflow,
+                            transValue(BC->getOperand(0), F, BB),
+                            transValue(BC->getOperand(1), F, BB)));
```

Comment:
This type needs to be i1 in the resulting LLVM IR. 
Can you please check?

Thanks

Reply:
This conversion already works by implicit conversion somewhere inside CreateBinaryIntrinsic. I'm not very fond of this kind of implicit behaviors and imo this should by default explode but it isn't. And now is a problem because without going inside CreateBinaryIntrinsic and finding out how conversion works there is no way to prove that forcing type of getOperand(1) works.
```
case OpIAddCarry: {
    IRBuilder Builder(BB);
    auto *BC = static_cast<SPIRVBinary *>(BV);
    SPIRVTypeBool BOC;
    BC->getOperand(1)->setType(&BOC);
    return mapValue(BV, Builder.CreateBinaryIntrinsic(
                            Intrinsic::uadd_with_overflow,
                            transValue(BC->getOperand(0), F, BB),
                            transValue(BC->getOperand(1), F, BB)));
  }
```
This would be a possible way to force type change.